src/sqlfluff/core/linter/common.py

```diff
@@ -65,6 +66,7 @@ class ParsedString(NamedTuple):
     templated_file: TemplatedFile
     config: FluffConfig
     fname: str
+    source_str: str
```

Comment:
Added `source_str` fields to these two structures so the linter has access to the raw source in case of template and/or parse failures. This allows the linter to find `noqa` comments even when these failures occur.